### ISSUE TITLE
cache clean doesn't clean everything in cache list

### ISSUE THREAD

**Issue by DrDaveD** [apptainer/singularity#2954]
## Version of Singularity:

Current master branch

### Expected behavior

`singularity cache clean` will clean up everything in `singularity cache list`

### Actual behavior

If there's an image in ~/.singularity/cache/oci-tmp, or ~/.singularity/cache/library, it is listed but not cleaned

### Steps to reproduce behavior

```
$ singularity exec docker://alpine id
...
$ singularity exec library://alpine id
...
$ singularity cache list
NAME                   DATE CREATED           SIZE             TYPE
alpine_latest.sif      2019-03-19 23:01:45    2.72 Mb          library
alpine_latest.sif      2019-03-19 23:00:29    2.72 Mb          oci
$ singularity cache clean
$ singularity cache list
NAME                   DATE CREATED           SIZE             TYPE
alpine_latest.sif      2019-03-19 23:01:45    2.72 Mb          library
alpine_latest.sif      2019-03-19 23:00:29    2.72 Mb          oci
```

@cclerget

**Comment by GodloveD** [comment]
By default, the `cache clean` command only cleans the blob cache.  You must pass other options to clean other parts of the cache.  In particular you can pass the `--all` option if you want to clean everything.  

I feel like this is pretty clear in the help, but I'm not sure if it is a sensible and intuitive default or not.  Maybe it would be better if `singularity cache clean` didn't do anything except for issue a message asking which cache to clean?  

What do you think @WestleyK ?

**Comment by DrDaveD** [comment]
Oh, I didn't think to try singularity cache clean --help, I only did singularity cache --help.  Yeah I think it would be better to not have a default to that command, let the default be to show the help.

Also I see a misspelling in that help, it says "spicific".

**Comment by WestleyK** [comment]
@GodloveD, Cleaning only the blob cache by default was @gmkurtzer decision. *The original behavior was to clean all cache by default.*

@DrDaveD, Is this #2972 what your thinking of by not clearing any cache by default?

**Comment by DrDaveD** [comment]
Yes, that's what I was thinking.  Make at least one of the options be required.

**Comment by DrDaveD** [comment]
Oh I see the typo was already fixed in #2960

**Comment by mem** [comment]
Behavior has been changed so that `cache clean` removes everything by default (with a confirmation prompt).
